gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java

```diff
@@ -0,0 +1,249 @@
+      return this;
+    }
+
+    @Override
+    public Builder configuration(DiskConfiguration configuration) {
+      infoBuilder.configuration(configuration);
+      return this;
+    }
+
+    @Override
+    public Builder diskId(DiskId diskId) {
+      infoBuilder.diskId(diskId);
+      return this;
+    }
+
+    @Override
+    Builder creationTimestamp(Long creationTimestamp) {
+      infoBuilder.creationTimestamp(creationTimestamp);
+      return this;
+    }
+
+    @Override
+    Builder creationStatus(CreationStatus creationStatus) {
+      infoBuilder.creationStatus(creationStatus);
+      return this;
+    }
+
+    @Override
+    public Builder description(String description) {
+      infoBuilder.description(description);
+      return this;
+    }
+
+    @Override
+    Builder licenses(List<LicenseId> licenses) {
+      infoBuilder.licenses(licenses);
+      return this;
+    }
+
+    @Override
+    Builder attachedInstances(List<InstanceId> attachedInstances) {
+      infoBuilder.attachedInstances(attachedInstances);
+      return this;
+    }
+
+    @Override
+    Builder lastAttachTimestamp(Long lastAttachTimestamp) {
+      infoBuilder.lastAttachTimestamp(lastAttachTimestamp);
+      return this;
+    }
+
+    @Override
+    Builder lastDetachTimestamp(Long lastDetachTimestamp) {
+      infoBuilder.lastDetachTimestamp(lastDetachTimestamp);
+      return this;
+    }
+
+    @Override
+    public Disk build() {
+      return new Disk(compute, infoBuilder);
+    }
+  }
+
+  Disk(Compute compute, DiskInfo.BuilderImpl infoBuilder) {
+    super(infoBuilder);
+    this.compute = checkNotNull(compute);
+    this.options = compute.options();
+  }
+
+  /**
+   * Checks if this disk exists.
+   *
+   * @return {@code true} if this disk exists, {@code false} otherwise
+   * @throws ComputeException upon failure
+   */
+  public boolean exists() {
+    return reload(DiskOption.fields()) != null;
+  }
+
+  /**
+   * Fetches current disk's latest information. Returns {@code null} if the disk does not exist.
+   *
+   * @param options disk options
+   * @return a {@code Disk} object with latest information or {@code null} if not found
+   * @throws ComputeException upon failure
+   */
+  public Disk reload(DiskOption... options) {
+    return compute.get(diskId(), options);
+  }
+
+  /**
+   * Deletes this disk.
+   *
+   * @return a zone operation if the delete request was successfully sent, {@code null} if the disk
+   *     was not found
+   * @throws ComputeException upon failure
+   */
+  public Operation delete(OperationOption... options) {
+    return compute.delete(diskId(), options);
+  }
+
+  /**
+   * Creates a snapshot for this disk given the snapshot's name.
+   *
+   * @return a zone operation if the snapshot creation was successfully requested. {@code null} if
+   *     the disk was not found
+   * @throws ComputeException upon failure
+   */
+  public Operation createSnapshot(String snapshot, OperationOption... options) {
```

Comment:
I think I missed that in the compute.create(SnapshotInfo). Any reason to change the pattern and return `null` in create in this case?